app/models/products/user_note_mode.rb

```diff
@@ -0,0 +1,49 @@
+module Products
+
+  class UserNoteMode
```

Comment:
I don't entirely grasp the raison d'être of this object. It's sort of a presenter and sort of a validator, and it's weird to me that it's in products and not in lib.  Can you explain why you made this choice?